src/pint/residuals.py

```diff
@@ -267,7 +270,23 @@ def get_PSR_freq(self, modelF0=True):
                     "No pulsar spin parameter(e.g., 'F0'," " 'P0') found."
                 )
             return F0.to(u.Hz)
-        else:
+        elif calctype.lower() == "taylor":
+            # see Spindown.spindown_phase
+            dt = self.model.get_dt(self.toas, 0)
+            # if the model is defined through F0, F1, ...
+            if "F0" in self.model.params:
+                fterms = [0.0 * u.dimensionless_unscaled] + self.model.get_spin_terms()
+
+            # otherwise assume P0, PDOT
+            else:
+                F0 = 1.0 / self.model.P0.quantity
+                if "PDOT" in self.model.params:
+                    F1 = -self.model.PDOT.quantity / self.model.P0.quantity ** 2
```

Comment:
I know this computation is sort of obvious, but it might be good form to use `derived_quantities.p_to_f()`